External/Plugins/FlashDebugger/Controls/WatchUI.cs

```diff
@@ -35,43 +41,55 @@ private void TreeControlResize(Object sender, EventArgs e)
             this.treeControl.Tree.Columns[1].Width = w - 8;
         }
 
-        public bool AddElement(string item)
+        private void WatchManager_ExpressionAdded(Object sender, WatchExpressionArgs e)
         {
-            if (watches.Contains(item)) return false;
-            watches.Add(item);
-            treeControl.Nodes.Insert(watches.Count - 1, GetExpressionNode(item));
+            treeControl.Nodes.Insert(e.Position, GetExpressionNode(e.Expression));
             UpdateElements();
-            return true;
+        }
+
+        private void WatchManager_ExpressionRemoved(Object sender, WatchExpressionArgs e)
+        {
+            UpdateElements();
+        }
+
+        private void WatchManager_ExpressionReplaced(Object sender, WatchExpressionReplaceArgs e)
+        {
+            treeControl.Nodes[e.Position] = GetExpressionNode(e.NewExpression);
```

Comment:
Well, the logic is centralized in the WatchManager instance, and I'd say it's not possible to break this part of the code unless some plugin developer makes some nasty stuff like accessing the private tree control through reflection, which AFAIK would not make sense having both the WatchUI and WatchManager instances accessible.

But I'd add some check.